### PR TITLE
Training

### PR DESCRIPTION
**Pull request by hazmat345**
We are creating a Beer Garden 101 course to help people learn how to create a Plugin and some of the cool features we have. This is heavily based off the User Guides with real world examples integrated into it.

### CODE REVIEW

File: docs/training/example-2.adoc [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+	client = NSLookupClient()
+	client.execute_nslookup("python.org")
+
+if __name__ == "__main__":
+	main()
+----
+
+Instead of converting the function directly into a Beer Garden System Command, let's build a new command
+that will pass in the domain being queried. That way we can expose this new function in the UI.
+
```
Comment (hazmat345) [inline docs/training/example-2.adoc:62]: I think this is not the best approach - one of the "best" things about brewtils is the minimal amount of changes that you need to make to convert an existing class to a brewtils plugin. Adding a new function that calls an existing one is more verbose and would double the amount of methods if applied to a "real" class. Also, having a method who's only statement is to call another method is not great practice.

I think adding the decorators to the existing methods is a better way.

File: docs/training/example-3.adoc [hunk 2]
```diff
@@ -0,0 +1,131 @@
+== Example 3: Web Scrapping
```
Comment (hazmat345) [inline docs/training/example-3.adoc:1]: Scraping

File: docs/training/example-3.adoc [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+this task in Python. For our examples we are going to use two additional libraries,
+link:https://requests.readthedocs.io/en/master/[Requests] and
+link:https://www.crummy.com/software/BeautifulSoup/[BeautifulSoup].
+
+Before we start, please ensure these libraries are installed:
+
+[source,bash,subs="attributes"]
+----
+pip install beautifulsoup4
+pip install requests
```
Comment (hazmat345) [inline docs/training/example-3.adoc:15]: This can just be

```
pip install beautifulsoup4 requests
```

File: docs/training/example-3.adoc [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+
+
+def main():
+    client = WebScrappingClient()
+
+    plugin = Plugin(
+        client,
+        name='WebScrapper',
+        version='1.0.0',
+        description='Scraps requested website with BS4',
```
Comment (hazmat345) [inline docs/training/example-3.adoc:97]: Scrapes

File: docs/training/example-3.adoc [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+        soup = self.download_url(url)
+        return soup.find(id=id).prettify()
+
+
+def main():
+    client = WebScrappingClient()
+
+    plugin = Plugin(
+        client,
+        name='WebScrapper',
```
Comment (hazmat345) [inline docs/training/example-3.adoc:95]: WebScraper

File: docs/training/training-101.adoc [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+
+include::ui-orientation.adoc[]
+
+== Writing Plugins for Beer Garden
+
+This is a guided plugin development that are intended to be deployed against an active Beer Garden
+instance. This only focuses on the developing remote plugins.
+
+link:../local-vs-remote.adoc[Local vs Remote]::
+If you are interested in the difference between Remote and Local plugins, check out our guide
```
Comment (hazmat345) [inline docs/training/training-101.adoc:40]: I'd honestly just take this part about local vs remote plugins out. I've always meant to move this out of the initial flow on the site. It's not something that needs to be noted until after people understand what a plugin really is. Focusing on it before actually doing the example is just going to confuse people and they're not going to retain the distinction.

File: docs/training/example-5.adoc [hunk 7]
```diff
@@ -1 +1,400 @@
-== Example 5: User Forms
+== Example 5: User Forms and Advance Features
```
Comment (hazmat345) [inline docs/training/example-5.adoc:1]: Advanced

File: docs/training/example-5.adoc [hunk 8]
```diff
@@ -1 +1,400 @@
...
+:includedir: _includes
+
+We have now covered the basic of how to create Beer Garden Plugins with a variety of Input Types and Output Types.
```
Comment (hazmat345) [inline docs/training/example-5.adoc:4]: basics

File: docs/training/example-5.adoc [hunk 9]
```diff
@@ -1 +1,400 @@
...
+So now lets run this Plugin and see what happens
+
+    python complex_beer_statistics_plugin.py
+
+image::{includedir}/complex-beer-statistics-test.png[Complex Beer Statistics Results]
+
+We can see that a child request was spawned when invoking this command. If you clicked on the hyperlink
+for `download_year_statistics` then you can see the input/output for that command.
+
+=== Building Dynamic Dropdown List
```
Comment (hazmat345) [inline docs/training/example-5.adoc:85]: If you specifically say "Dropdown" you should add that to the `choices` definition, right now they are typeaheads.

**Comment by TheBurchLog** [comment]
For this training I was going with a more informal approach. When creating the outline in my head I was thinking

- Give overview of what is Beer Garden and how it can help
- Provide some orientation on the UI that will be helpful later
- Examples
  - Most basic Hello World
  - Showing how they can run their bash commands with Beer Garden
  - Showing how to render HTML
  - Showing how to render JSON
  - Showing how to link Plugins together and use advance features of Forms
- Try to get them to continue using BG or Contribute

After every example I wanted to give them examples on how they could use this for other projects. 

When giving this class in person, this Training 101 is more of the voice over I would be providing. Almost like a script that any instructor could read. That way a person could take this in person or on their own and get a similar experience.

**Comment by TheBurchLog** [comment]
Based off our conversation, I am going to merge in the training. It will not have a reference link from the website, so users will need a direct link to access it.

**Pull request merged**